Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -18,9 +18,14 @@ public protocol NavigationService: CLLocationManagerDelegate, RouterDataSource,
     var locationManager: NavigationLocationManager { get }
     
     /**
-     A reference to a MapboxDirections service. Used for rerouting.
+     `NavigationProvider`, used to create route.
+     */
+    var routingProvider: NavigationProvider { get }
+    
+    /**
+     Credentials data, used to authorize server requests.
      */
-    var directions: Directions { get }
```

Comment:
The removal of this property spurs a bunch of changes to MapboxNavigation, such as in NavigationViewController and JunctionView, which makes me a bit concerned about backwards compatibility and thus eligibility for v2._x_. What if we retain this property but mark it as deprecated and document that it is ignored for rerouting purposes? We can introduce a parallel property of the new, more general type to use as a stand-in.

Reply:
I think leaving such dangling property may be even more confusing. Compiler won't complain about something missing, but any interactions with it will be just silently ignored which will most likely result in bugs in user apps.
On the other hand, replacing it with new property allows straightforward migration. If user stays on default settings for routing, he'll just need to cast the object to `Directions` and continue his previous flow.